client/src/Counter.js

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react"
+const Counter = ({ setAllMyVideos, videoId, videoRating }) => {
+    const [rating, setRating] = useState(0)
+
+    const handleCounterDown = () => {
+
+        if (rating <= 0) {
+            setRating(0)
+            handleRatingChange(rating)
```

Comment:
`setRating` doesn't actually update `rating` right away, it schedules it to be updated in the future - if you `console.log` it you should see it still has the old value - you may want to set a variable you can use here, e.g.
```javascript
const newRating = rating <= 0 ? 0 : rating - 1;
handleRatingChange(newRating);
```

You may see at the moment that your app is one click behind on rating updates :)

Reply:
Updated counter, working a little bit better, not perfect, but better